### PR TITLE
 Fixes for RC commands 

### PR DESCRIPTION
**Pull request by LitvinenkoIra**

### CODE REVIEW

**Comment by LitvinenkoIra** [comment]
@BSolonenko please review

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/include/rc_rpc_plugin/commands/hmi/rc_on_interior_vehicle_data_notification.h [hunk 1]
```diff
@@ -22,9 +22,10 @@ class RCOnInteriorVehicleDataNotification
 
   ~RCOnInteriorVehicleDataNotification();
 
-  bool Init() OVERRIDE;
-  void Run() OVERRIDE;
-  void on_event(const application_manager::event_engine::Event& event);
+  /**
+   * @brief Execute command
+   **/
+  virtual void Run();
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/rpc_plugins/rc_rpc_plugin/include/rc_rpc_plugin/commands/hmi/rc_on_interior_vehicle_data_notification.h]: @LitvinenkoIra avoid virtual and add override

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/include/rc_rpc_plugin/commands/mobile/set_interior_vehicle_data_request.h [hunk 2]
```diff
@@ -71,10 +71,16 @@ class SetInteriorVehicleDataRequest : public RCCommandRequest {
...
 
+  std::string ModuleType() FINAL;
+
   /**
    * @brief SetInteriorVehicleDataRequest class destructor
    */
   ~SetInteriorVehicleDataRequest();
+
+private:
+  const smart_objects::SmartObject& ControlData(
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/rpc_plugins/rc_rpc_plugin/include/rc_rpc_plugin/commands/mobile/set_interior_vehicle_data_request.h]: @LitvinenkoIra description missed

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/hmi/rc_on_remote_control_settings_notification.cc [hunk 3]
```diff
@@ -98,12 +98,10 @@ void RCOnRemoteControlSettingsNotification::Run() {
...
           (*message_)[app_mngr::strings::msg_params]
-                     [message_params::kAccessMode].asString();
-
-      access_mode = AccessModeFromString(access_mode_str);
-      LOG4CXX_DEBUG(logger_, "Setting up access mode : " << access_mode_str);
+                     [message_params::kAccessMode].asUInt());
+      LOG4CXX_DEBUG(logger_, "Setting up access mode : " << AccessModeToString(access_mode));
     } else {
       access_mode = resource_allocation_manager_.GetAccessMode();
       LOG4CXX_DEBUG(logger_,
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/hmi/rc_on_remote_control_settings_notification.cc]: @LitvinenkoIra extract logging from 2 branches of if

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/mobile/button_press_request.cc [hunk 4]
```diff
@@ -148,13 +149,13 @@ bool CheckButtonName(const std::string& module_type,
...
   LOG4CXX_AUTO_TRACE(logger_);
 
-  const std::string button_name =
-      (*message_)[app_mngr::strings::msg_params][message_params::kButtonName]
-          .asString();
-  const std::string module_type =
-      (*message_)[app_mngr::strings::msg_params][message_params::kModuleType]
-          .asString();
+  const char* button_name;
+  NsSmartDeviceLink::NsSmartObjects::EnumConversionHelper
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/mobile/button_press_request.cc]: @LitvinenkoIra such function should be extracted to something like it https://github.com/LuxoftAKutsan/sdl_core/blob/42fdf034508100c756e68c936fa21d4dc7626d5b/src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/resource_allocation_manager_impl.cc#L216 
May be in message helper

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/mobile/button_press_request.cc [hunk 5]
```diff
@@ -170,17 +171,26 @@ void ButtonPressRequest::Execute() {
       rc_capabilities &&
       CheckIfButtonExistInRCCaps(*rc_capabilities, button_id);
 
+  smart_objects::SmartObject msg_params =
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/mobile/button_press_request.cc]: @LitvinenkoIra do we need copy of message params?

**Review by LuxoftAKutsan — APPROVED** [review]

**Pull request merged**